### ISSUE TITLE
qsfs tutorial

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/home#1539]
https://git.ourworld.tf/tfgrid/circle_engineering/issues/23

- [x] how to setup
- [x] how to connect to created ZDB's using TFCMD
- [ ] how to mount QSFS
- [ ] vscript example to setup a filesystem using ZDB's all in UBuntu 23.10 using vlang vscript
- [ ] how to see see monitoring/health
- [ ] check that we can repair, how can we see we need repair
- [ ] check that we can re-connect on other VM
- [ ] make a test script where we destroy a VM which was writing to QSFS, then create new VM which remounts QSFS, there might be some data lost, but never longer than e.g. 15 min
- [ ] check that all metadata is there as needed to reconnect, document where metadata is
- [ ] explain architecture and how we use local ZDB as cache

#### repo:
 - https://github.com/threefoldtech/quantum-storage

#### Testsuite to be reviewed:
  - https://app.testlodge.com/a/26076/projects/40893/suites/195671

**Comment by ramezsaeed** [comment]
@LeeSmet 
can you please update us here by what is implemented and what is not yet so we can start testning, and @Mik-TF can start doing the Docs for that.

**Comment by ramezsaeed** [comment]
Currently we have these tutorial:
https://manual.grid.tf/documentation/system_administrators/terraform/resources/terraform_qsfs.html

**Comment by A-Harby** [comment]
What I found so far:
- For terraform:
  - The [docs](https://manual.grid.tf/documentation/system_administrators/terraform/resources/terraform_qsfs.html).
  - The [example](https://github.com/threefoldtech/terraform-provider-grid/tree/development/examples/resources/qsfs), but it needs to be updated to work fine, and I reported it [here](https://github.com/threefoldtech/terraform-provider-grid/issues/948) but I was able to get the code to work in the end.
- For TS client:
  - The [docs](https://manual.grid.tf/documentation/developers/javascript/grid3_javascript_qsfs.html).
  - The [example](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/grid_client/scripts/vm_with_qsfs.ts).

**Comment by Mik-TF** [comment]
Great! Thanks for this research and testing.

With terraform, you got it to work finally? If yes can you share the files you used? Thanks.

**Comment by A-Harby** [comment]
> Update:
> 
> * Found script for QSFS in the Go client, but it is written as a [test](https://github.com/threefoldtech/tfgrid-sdk-go/edit/development/grid-client/integration_tests/qsfs_test.go).
> * Didn't find way to deploy in tfcmd and no docs for it.

I created an issue https://github.com/threefoldtech/tfgrid-sdk-go/issues/1103 to add a way to deploy QSFS in TFCMD.

**Comment by ramezsaeed** [comment]
@LeeSmet 
can you please check this one as its important for @Mik-TF  to complete the manual.

https://github.com/threefoldtech/quantum-storage/issues/38

**Comment by A-Harby** [comment]
I created this issue (https://github.com/threefoldtech/quantum-storage/issues/39) to create a bash script to make the installation easier for the user.

And I am still trying to get qsfs to work following these two files: https://github.com/threefoldtech/info_grid/pull/588/files and https://github.com/threefoldfoundation/info_threefold_pub/blob/main/wiki/qss/manual/qsfs_setup.md.

**Comment by Mik-TF** [comment]
Scott has some work done on info_grid. It might be of use. https://github.com/threefoldtech/info_grid/pull/588

**Comment by A-Harby** [comment]
Updates:

I have reviewed the PR and found some comments on the quantum_safe_storage_deploy.md file that need to be fixed or to have more explanation. https://github.com/threefoldtech/info_grid/pull/588.

And I have also followed all the steps to deploy QSFS, but I am still failing to get it to work and am still debugging.

```
root@vmjlr69:~# zstor --log_file ~/zstor.log -c /etc/zstor-default.toml status
Error: ZstorError { kind: Config, internal: Other(Error { inner: ErrorInner { kind: Wanted { expected: "a right bracket", found: "an identifier" }, line: Some(28), col: 19, at: Some(586), message: "", key: [] } }) }
```

**Comment by A-Harby** [comment]
I have added some more comments to the [PR](https://github.com/threefoldtech/info_grid/pull/588).

And I created some issues that I found:
- https://github.com/threefoldtech/quantum-storage/issues/40
- https://github.com/threefoldtech/0-db/issues/167
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/1128

**Comment by Mik-TF** [comment]
Amazing @A-Harby, you're on fire! Will check the info_grid PR with Scott. Thanks.

**Comment by A-Harby** [comment]
Updates:
I created a new issue for the cache: https://github.com/threefoldtech/quantum-storage/issues/41.

All of the tests can't be done until all of the issues are resolved. https://github.com/threefoldtech/home/issues/1539#issuecomment-2242913928. 

And I am still waiting for the fixes in the PR https://github.com/threefoldtech/info_grid/pull/588/files#diff-b0d52cfbe1929e80b61aa5d44f601ec5487d3ec6e5cf228cad7ed7af47aeea97 to check docs boxes.

**Comment by A-Harby** [comment]
> * [x]  how to setup
> * [x]  how to connect to created ZDB's using TFCMD

> * [ ]  how to mount QSFS
> * [ ]  check that all metadata is there as needed to reconnect, document where metadata is
> * [ ]  explain architecture and how we use local ZDB as cache
> * [ ]  how to see see monitoring/health
> * [ ]  check that we can repair, how can we see we need repair
> * [ ]  check that we can re-connect on other VM

These should be covered here [PR](https://github.com/threefoldtech/info_grid/pull/588/files#diff-b0d52cfbe1929e80b61aa5d44f601ec5487d3ec6e5cf228cad7ed7af47aeea97).

> * [ ]  make a test script where we destroy a VM which was writing to QSFS, then create new VM which remounts QSFS, there might be some data lost, but never longer than e.g. 15 min

This test and test suite will be created after the issues are solved. https://github.com/threefoldtech/home/issues/1539#issuecomment-2242913928.

> * [ ]  vscript example to setup a filesystem using ZDB's all in UBuntu 23.10 using vlang vscript

There is no possible, clear way to do that yet.

**Comment by A-Harby** [comment]
Blocked on https://github.com/threefoldtech/home/issues/1539#issuecomment-2242913928, and @scottyeager confirmed the issue https://github.com/threefoldtech/quantum-storage/issues/41 and also found another bug https://github.com/threefoldtech/0-stor_v2/issues/122.

**Comment by mik-tf** [comment]
For now we won't focus on QSFS.
